src/borg/testsuite/hashindex_pytest.py

```diff
@@ -56,3 +56,13 @@ def test_hashindex_compact():
     for k, v in kv.items():
         assert k in idx and idx[k] == (v, v, v)
     assert len(idx) == len(kv)
+    # now expand the hashtable again. trigger a resize/rebuild by adding an entry.
+    k = b"x"*32 # list(kv)[0]
+    idx[k] = (0, 0, 0)
```

Comment:
hangs here. no output, no boom, no ctrl-c. why?

i searched the code for NDEBUG, but it is used nowhere, so shouldn't the assert already do something?

why doesn't even the null pointer deref terminate the program?

Reply:
also:
- there is no empty or deleted bucket in the compacted hashtable
- there is no matching key in the hashtable

thus i am rather sure we should run into the "BOOM!".

Reply:
it now segfaults, so it gets there.